src/event/cloudevent.ts

```diff
@@ -165,4 +169,8 @@ export class CloudEvent implements CloudEventV1, CloudEventV03 {
       }
     }
   }
+
+  public cloneWith(options?: any): CloudEvent {
```

Comment:
You should type this parameter like this:

```suggestion
  public cloneWith(event: CloudEventV1 | CloudEventV1Attributes | CloudEventV03 | CloudEventV03Attributes): CloudEvent {
```

Reply:
yea,  it was a placeholder for the moment.  But i don't think this should be a CloudEvent Type.  The value passed in could just be a couple extensions like this : `{ ext1: "value1", ext2: "value2" }`  and it Typscript will error in that case.

Reply:
reading what i wrote,  i don't think i explained it well enough.

I think we should be able to do this:

```
// Initial Creation
let cloudevent = new CloudEvent({....});

// Update with some extensions
cloudevent = cloudevent.cloneWith({ ext1: "value1", ext2: "value2"  });
```

You can't do that update if the function parameter is of the CloudEvent type.   I was assuming this was how it should work,  you pass in the things you want to add.

Reply:
just updated with the suggestions we talked about